go/fn/adaptor.go

```diff
@@ -0,0 +1,35 @@
+package fn
```

Comment:
Sg!

Reply:
> > Reduce the exposed surface: moving variants to internal.
> 
> The `variants` were private so they are not exposed surface. From a user's perspective, there are no surface changes related to `variants`.

Yeah, that's true. I should say avoid exposing `kustomize/kyaml`. 😄  I actually want to achieve two goals when moving variants to ./internal:
1. Users won't import or realize the dependency of  `kustomize/kyaml`  under the hood. (It reminds me that I should move adaptor to a different package since it breaks this rule.) So that we won't worry about backward compatibility when fork our own kustomize/kyaml.
2. Users won't be confused by the many go files that contain  private-only functions like `./fn/variant.go` `./fn/map.go` `./fn/slice.go`. When exploring the ./go/fn directory, they get a clean overview about the available functions and structs.